server/services/schedule-service.js

```diff
@@ -2,7 +2,22 @@ const scheduleRepository = require('../repository/schedule-repository')
 
 const BASE_PAYMENTS = 6
 
+function aggregateByClaimId (scheduleLines) {
+  const schedules = scheduleLines.reduce((r, a) => {
+    r[a.claimId] = [...r[a.claimId] || [], a.paymentDate]
+    return r
+  }, {})
+
+  return Object
+    .entries(schedules)
+    .map((e) => { return { claimId: e[0], paymentDates: e[1] } })
+}
+
 module.exports = {
+  getAll: async function () {
+    const scheduleLines = await scheduleRepository.getAll()
+    return aggregateByClaimId(scheduleLines)
```

Comment:
Would this be better done in SQL rather than a getAll passed to node, i.e.
```
SELECT "claimId", array_to_string(array_agg("paymentDate"), ',')
FROM schedules
GROUP BY "claimId";
```

Reply:
The change to using the ORM means the aggregation is no longer needed.